python_benchmarks/core.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import ctypes
+import torch
+from torch.profiler import profile, ProfilerActivity
+
+libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+for libname in libnames:
+    try:
+        cuda = ctypes.CDLL(libname)
+    except OSError:
+        continue
+    else:
+        break
+else:
+    raise OSError("could not load any of: " + " ".join(libnames))
+
+CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE = 38
```

Comment:
I'd at least putting enough comments/references to explain this magic number. I'm quite nervous about future drivers breaking this assumption in the future. Won't 38 potentially change when the driver is upgraded?

Reply:
It's a possibility. Historically, they don't seem to have changed. I'll add the links to the code.

https://nvidia.github.io/cuda-python/module/cuda.htmlhighlight=l2+cache+size#cuda.cuda.CUdevice_attribute.CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE